x-pack/legacy/plugins/uptime/public/lib/compose/kibana_compose.ts

```diff
@@ -4,13 +4,13 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import uiRoutes from 'ui/routes';
-import { UMKibanaFrameworkAdapter } from '../adapters/framework/kibana_framework_adapter';
+import { npStart } from 'ui/new_platform';
```

Comment:
Since in the NP this will only be available in the Plugin's `start()` function it would be good to move this import into `index.ts` and pass it as an argument into a new `Plugin` class like you did with the server-side shim.

Reply:
Can you have a look at https://github.com/elastic/kibana/pull/44722/commits/0bc73954c1227cc295e045f67f5e73e63c283c57 and let me know if we're a little closer on this?

Reply:
Left a comment on injecting coreStart into `start()`, but otherwise this looks good and injecting it into `getKibanaFrameworkAdapter()` is exactly what I had in mind.

Reply:
++

This file is actually outdated, the libs are being bootstrapped from the `new_platform_adapter.ts` file. But the dependencies are injected in the same way there, and that function is called as part of our new `plugin`'s `start` function.